### PR TITLE
Use one pod.yml file; two images; func test script

### PR DESCRIPTION
**Pull request by portante**
We now have one `pod.yml` file instead of two with almost identical contents. This necessitates moving the CI pipeline image names to `jenkins/Pipeline.gy`.

We now have two images we build: (new) a stand-alone Pbench Server image, and a the original "Pbench-in-a-Can" image used by the CI. The stand-alone image is built first, and then used by the CI image build to construct that iamge.

Finally, we provide a wrapper script, `jenkins/run-server-func-tests`, for running server functional tests to allow for sane defaults to be chosen (`envsubst` does not allow for that).

Tested locally with a script like:

```
make -C server/rpm clean rpm

export PB_CONTAINER_REG=<some registry>
export PB_ORG_NAME=pbench
export PB_SERVER_IMAGE_NAME=pbench-server-ci
export PB_SERVER_IMAGE_TAG=portante

RPM_PATH=${HOME}/rpmbuild/RPMS/noarch/pbench-server-*.rpm bash -ex server/pbenchinacan/container-build-ci.sh
buildah push localhost/${PB_SERVER_IMAGE_NAME}:${PB_SERVER_IMAGE_TAG} ${PB_CONTAINER_REG}/${PB_ORG_NAME}/${PB_SERVER_IMAGE_NAME}:${PB_SERVER_IMAGE_TAG}

jenkins/run-server-func-tests --cleanup
```

### CODE REVIEW

**Review by portante — COMMENTED** [review]
Ready for review -- pulled from PR #3081 to isolate the changes.

**Review by dbutenhof — CHANGES_REQUESTED** [review]
Mostly great. There's a comment to be fixed, and I think the `--cleanup` behavior on the functional test wrapper should be tweaked.

File: jenkins/runlocal [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+export PB_ORG_NAME=pbench
+export PB_SERVER_IMAGE_NAME=pbench-server-ci
+export PB_SERVER_IMAGE_TAG=${USER}
+export PB_SERVER_IMAGE="${PB_INTERNAL_CONTAINER_REG}/${PB_ORG_NAME}/${PB_SERVER_IMAGE_NAME}"
+export PB_POD_NAME=pbench-in-a-can
+export PB_DASHBOARD_DIR=$(pwd)/dashboard/build
+
+make -C server/rpm clean rpm
+
+#buildah login -u='$app' -p="8YSHSK5DO1CWUY882NFGHDEC5Q0VY0ZFYNGD3M99UR6EC8XHDWR9ZFOHI108SYWIC359MSFUIFAKRZADPEM9BZQOU2VGUB4EX3RLD4IOYJN4CT6VVZ6NQOOC" ${PB_INTERNAL_CONTAINER_REG}
```
[the remark this answers is not in the document]
Comment (portante) [reply jenkins/runlocal]: Bugger.  Fixed.

File: jenkins/run-server-func-tests [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+podman pod stop ${PB_POD_NAME}
+stop_status=${?}
+if [[ ${exit_status} -eq 0 ]]; then
+    exit_status=${stop_status}
+fi
+podman pod rm ${PB_POD_NAME}
+stop_status=${?}
+if [[ ${exit_status} -eq 0 ]]; then
+    exit_status=${stop_status}
+fi
```
[the remark this answers is not in the document]
Comment (portante) [reply jenkins/run-server-func-tests:86]: Fixed.

File: jenkins/run-server-func-tests [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    curl ${SERVER_API_ENDPOINTS}
+    exit 1
+fi
+
+EXTRA_PODMAN_SWITCHES="${EXTRA_PODMAN_SWITCHES} --network host" jenkins/run server/exec_functests ${SERVER_URL}
+exit_status=${?}
+if [[ ${exit_status} -ne 0 ]]; then
+    # Try to capture the functional test container's log on completion.
+    # On success, the pod has already been shut down, so ignore errors.
+    # TODO: is there a more elegant way to do this?
```
[the remark this answers is not in the document]
Comment (portante) [reply jenkins/run-server-func-tests]: Fixed.

[the remark this answers is not in the document]
Comment (dbutenhof) [reply jenkins/run-server-func-tests]: Well, you removed the TODO, but the rest seems to remain and the pod is *not* shut down at this point...

[the remark this answers is not in the document]
Comment (portante) [reply jenkins/run-server-func-tests]: Oops.  Fixed.

**Review by dbutenhof — APPROVED** [review]

File: jenkins/run-server-func-tests [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+echo "Waiting for pod's httpd to show up ..."
+until curl -s -o /dev/null ${SERVER_API_ENDPOINTS}; do
+    sleep 1
+done
+echo "Waiting for pod's Pbench Server to show up ..."
+status_code=$(curl -s -o /dev/null -w "%{http_code}" ${SERVER_API_ENDPOINTS})
+while [[ "${status_code}" == "503" ]]; do
+    sleep 1
+    status_code=$(curl -s -o /dev/null -w "%{http_code}" ${SERVER_API_ENDPOINTS})
+done
```
Comment (npalaska) [inline jenkins/run-server-func-tests:56]: No counter to break the loop?

**Review by ndokos — DISMISSED** [review]
LGTM

File: jenkins/run-server-func-tests [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+podman pod stop ${PB_POD_NAME}
+stop_status=${?}
+if [[ ${exit_status} -eq 0 ]]; then
+    exit_status=${stop_status}
+fi
+podman pod rm ${PB_POD_NAME}
+stop_status=${?}
+if [[ ${exit_status} -eq 0 ]]; then
+    exit_status=${stop_status}
+fi
```
Comment (npalaska) [inline jenkins/run-server-func-tests:85]: If the pod is run without `--cleanup` then with this code the pod will still be cleaned up at the end of the `run-server-func-tests`, is that intentional?

**Review by npalaska — APPROVED** [review]
Just a couple of questions.

File: jenkins/run-server-func-tests (hunk 4 above)
Comment (portante) [reply jenkins/run-server-func-tests]: Yes, we probably should calculate some upper bound for a timeout.  Fixed.

**Review by dbutenhof — APPROVED** [review]

**Review by ndokos — APPROVED** [review]

**Review by npalaska — APPROVED** [review]

**Pull request merged**